plugin.jest.js

```diff
@@ -136,14 +137,55 @@ describe('CspHtmlWebpackPlugin', () => {
         const expected =
           "base-uri 'self';" +
           " object-src 'none';" +
-          " script-src 'unsafe-inline' 'self' 'unsafe-eval' 'sha256-ixjZMYNfWQWawUHioWOx2jBsTmfxucX7IlwsMt2jWvc=' 'nonce-mockedbase64string-1' 'nonce-mockedbase64string-2';" +
+          " script-src 'unsafe-inline' 'self' 'unsafe-eval' 'sha256-ixjZMYNfWQWawUHioWOx2jBsTmfxucX7IlwsMt2jWvc=' 'sha256-J7wa7HNc5Nb9SvdpRj1UEzZlXOJERU6Mw8r5DSsL1Go=' 'nonce-mockedbase64string-1' 'nonce-mockedbase64string-2';" +
           " style-src 'unsafe-inline' 'self' 'unsafe-eval' 'sha256-MqG77yUiqBo4MMVZAl09WSafnQY4Uu3cSdZPKxaf9sQ=' 'nonce-mockedbase64string-3'";
 
         expect(csps['index.html']).toEqual(expected);
         done();
       });
     });
 
+    it('inserts hashes for linked scripts and styles from the same Webpack build', (done) => {
```

Comment:
Could we add another test which checks that external scripts that are not generated by this webpack compilation don't have a hash added to the csp?

Could we add another test that ensures that if a single webpack compilation generates multiple (2 will do) instances of HtmlWebpackPlugin, each with different assets in them, the correct hashes are only added to the correct meta tags?
e.g.:
* webpack entry point `index-1.js`.
  * Generates HTML: `index-1.html` that has an external link `index-1.bundle.js` and a csp hash for `index-1.bundle.js`
* webpack entry point `index-2.js`.
  * Generates HTML: `index-2.html` that has an external link `index-2.bundle.js` and a csp hash for `index-2.bundle.js`

Reply:
added: https://github.com/slackhq/csp-html-webpack-plugin/pull/87/files#diff-fc33293bee015873cf89f2cff90af742b95c8a5117d0ee47f75f8ba2dcd4579bR189 (`it('only inserts hashes for linked scripts and styles from the same HtmlWebpackPlugin instance'`)